nixos/modules/system/boot/loader/raspberrypi-uboot/builder.sh

```diff
@@ -0,0 +1,24 @@
+#! @bash@/bin/sh -e
+
+# Delete old boot folder content
+rm -rf /boot/*
```

Comment:
Oh, that looks a bit scary. Can you make updating the boot files a bit more "atomic". (Disclaimer: I haven't looked at how other raspi distros do this.)

Reply:
Yeah, I also thought about that. I could first write all the files to a temporary directory and then swap these directories. Or do you have any better idea.

Reply:
If that works, perfect. But /boot might be a mountpoint, right? That may complicate things.

I don't have a concrete idea. However, one I'd like to say that perhaps a trap can be used for some added robustness. For example, if the original files are first moved to a .bak directory, and the trap restores from .bak in case of error.

Reply:
Why not move the old files to `/boot/old` or similar like it is done in [raspberrypi loader](https://github.com/bkchr/nixpkgs/blob/f49357cb5cddb545ff8d64132fae3723f8a3d41c/nixos/modules/system/boot/loader/raspberrypi/builder.sh#L16)?